File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/player/PlayerState.java
package com.github.anrimian.musicplayer.domain.models.player;

/**
 * Created on 02.11.2017.
 */

public enum PlayerState {
    IDLE,
    PLAY,
    PAUSE,
    STOP,
    LOADING,
    PAUSED_EXTERNALLY {

        @Override
        public PlayerState toBaseState() {
            return PAUSE;
        }
    };

    public PlayerState toBaseState() {
        return this;
    }
}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/player/PlayerState.kt
package com.github.anrimian.musicplayer.domain.models.player

sealed interface PlayerState {
    data object IDLE: PlayerState
    data object LOADING: PlayerState
    data object PREPARING: PlayerState
    data object PLAY: PlayerState
    data object PAUSE: PlayerState
    data object STOP: PlayerState
    data class Error(val throwable: Throwable): PlayerState
}